src/Orders/PurchaseUnit.php

```diff
@@ -36,9 +36,9 @@ class PurchaseUnit implements Arrayable, Jsonable
     /**
      * Create a new collection.
      */
-    public function __construct(string $currency_code, float $value)
+    public function __construct(string $currency_code, float $value, float $item_total, float $discount= 0.0)
```

Comment:
@abdelwahidJobs I think we should not change the API, but instead, allow user to overide the amount breakdown. something like this:

```php
$purchase_unit->setAmount(new AmountBreakdown($currency_code, $value, $item_total, $discount));
```

Reply:
cc @abdelwahidJobs

Reply:
@mohammed-elhaouari  that sound good

Reply:
@mohammed-elhaouari  I'm thinking about them all 
`shipping` , `handling` , `tax_total`, `insurance`, `shipping_discount`, `discount`
the AmountBreakdown should contain a collection of amounts/ items  because each one of these items is an `Object`
``` PHP 
        $this->amount = new AmountBreakdown($currency_code, $total);
        $this->amount->addShipping(200);
        $this->amount->addDiscount(200);
        $this->amount->addHandling(200);
        $this->amount->addTaxTotal(200);
        $this->amount->addInsurance(200);
        $this->amount->addShippingDiscount(200);
``` 
This way we can add any new object if we need

Reply:
Yes, this can do the trick. it's better.